network/tests/test_network.py

```diff
@@ -0,0 +1,234 @@
+# (C) Datadog, Inc. 2010-2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+from collections import namedtuple
+import socket
+import os
+import platform
+
+# project
+from datadog_checks.network import Network
+
+import mock
+import pytest
+
+FIXTURE_DIR = os.path.dirname(__file__)
+
+CX_STATE_GAUGES_VALUES = {
+    'system.net.udp4.connections': 2,
+    'system.net.udp6.connections': 3,
+    'system.net.tcp4.established': 1,
+    'system.net.tcp4.opening': 0,
+    'system.net.tcp4.closing': 0,
+    'system.net.tcp4.listening': 2,
+    'system.net.tcp4.time_wait': 2,
+    'system.net.tcp6.established': 1,
+    'system.net.tcp6.opening': 0,
+    'system.net.tcp6.closing': 1,
+    'system.net.tcp6.listening': 1,
+    'system.net.tcp6.time_wait': 1,
+}
+
+network_check = Network('network', {}, {})
+
+
+@pytest.fixture
+def ss_subprocess_mock(*args, **kwargs):
+    if args[0][-1] == '-4' and args[0][-3] == '-u':
+        file_name = 'ss_ipv4_udp'
+    elif args[0][-1] == '-4' and args[0][-3] == '-t':
+        file_name = 'ss_ipv4_tcp'
+    elif args[0][-1] == '-6' and args[0][-3] == '-u':
+        file_name = 'ss_ipv6_udp'
+    elif args[0][-1] == '-6' and args[0][-3] == '-t':
+        file_name = 'ss_ipv6_tcp'
+
+    with open(file_name, FIXTURE_DIR) as f:
+        contents = f.read()
+        contents = contents.decode('string-escape')
+        return contents.decode("utf-8")
+
+
+@pytest.fixture
+def netstat_subprocess_mock(*args, **kwargs):
+    if args[0][0] == 'ss':
+        raise OSError
+    elif args[0][0] == 'netstat':
+        with open('netstat', FIXTURE_DIR) as f:
+            contents = f.read()
+            contents = contents.decode('string-escape')
+            return contents.decode("utf-8")
+
+
+@pytest.fixture
+def aggregator():
+    from datadog_checks.stubs import aggregator
+    aggregator.reset()
+    return aggregator
+
+
+@pytest.mark.skipif(platform.system() is not 'Linux',
+                    reason="Only runs on Unix systems")
+@mock.patch('datadog_checks.network.network.get_subprocess_output', side_effect=ss_subprocess_mock)
+@mock.patch('datadog_checks.network.network.Platform.is_linux', return_value=True)
+def test_cx_state_linux_ss(aggregator, mock_subprocess):
+    network_check.check({})
+
+    # Assert metrics
+    for metric, value in CX_STATE_GAUGES_VALUES.iteritems():
+        aggregator.assert_metric(metric, value=value, tags=['optional:tag1'])
+
+
+@pytest.mark.skipif(platform.system() is not 'Linux',
+                    reason="Only runs on Unix systems")
+@mock.patch('datadog_checks.network.network.get_subprocess_output', side_effect=netstat_subprocess_mock)
+@mock.patch('datadog_checks.network.network.Platform.is_linux', return_value=True)
+def test_cx_state_linux_netstat(self, *args):
```

Comment:
what does `self` contain?

Reply:
Missed changing that since the test doesn't run locally. The args to this function should actually be: the list of the mock decorators, from bottom up, then whatever pytest fixtures are needed. Changing that.